torch/csrc/distributed/autograd/utils.cpp

```diff
@@ -78,8 +80,22 @@ Message getMessageWithProfiling(
     torch::distributed::rpc::Message&& wrappedRpcMessage,
     MessageType msgType,
     torch::autograd::profiler::ProfilerConfig&& profilerConfig) {
+  auto& remoteProfilerManager =
+      torch::distributed::rpc::RemoteProfilerManager::getInstance();
+  auto key = remoteProfilerManager.getCurrentProfilingKey();
+  // generate a globally unique Id
+  auto localId = remoteProfilerManager.getNextLocalId();
+  auto localWorkerId = RpcAgent::getCurrentRpcAgent()->getWorkerInfo().id_;
+  auto globallyUniqueId =
+      torch::distributed::rpc::ProfilingId(localWorkerId, localId);
```

Comment:
Ideally, we should have an API like `GloballyUniqueId::getNextId()`, which looks up worker id from RpcAgent and has its own global next local id to use for generation. This way we don't repeat the same boilerplate in multiple places to generate a unique id. Doesn't have to be part of this PR, but would be a good future refactor

Reply:
If `GloballyUniqueId` itself starts to generate the next ID by increment it's own internal counter, we may lose the convenient property that the IDs always increase by 1, i.e. the RRefIds currently would be (0, 1), (0, 2), ... and the dist autograd ID would be (workerId << 48), (workerId << 48) + 1, and so on. I don't think we rely on this anywhere, but it may be useful to keep this for debugging purposes?